src/app.js

```diff
@@ -1,3 +1,48 @@
 'use strict';
 
-// Write your code here
+const readline = require('readline');
+const { checkIsValidUserInput } = require('./modules/checkIsValidUserInput');
+const { generateRandomNumber } = require('./modules/generateRandomNumber');
+const { getBullsAndCows } = require('./modules/getBullsAndCows');
+
+const terminal = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+let userInput = null;
+
+function askForNumber() {
+  return new Promise((resolve) => {
+    terminal.question('Enter a 4-digit(unique) number: ', (number) => {
+      if (checkIsValidUserInput(number)) {
+        userInput = number;
+      } else {
+        terminal.write('Invalid input\n');
```

Comment:
would be great to specify what the user has done wrong. You can do it inside the validation function